modules/core/src/main/scala/org/tessellation/infrastructure/rewards/DTM.scala

```diff
@@ -0,0 +1,51 @@
+package org.tessellation.infrastructure.rewards
+
+import cats.data.NonEmptyMap
+
+import scala.concurrent.duration.FiniteDuration
+
+import org.tessellation.config.types.DTMConfig
+import org.tessellation.domain.rewards._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric._
+
+object DTM {
+
+  def make(config: DTMConfig, timeTriggerInterval: FiniteDuration): DTM = new DTM {
+
+    def getAddress: Address = config.address
+
+    def weight(
+      ignore: Set[Address]
+    )(distribution: NonEmptyMap[Address, Amount]): Either[RewardsError, NonEmptyMap[Address, Amount]] =
+      for {
+        withoutIgnored <- NonEmptyMap
+          .fromMap(ignore.foldLeft(distribution.toSortedMap) {
+            case (acc, addressToIgnore) => acc - addressToIgnore
+          })
+          .toRight[RewardsError](IgnoredAllAddressesDuringWeighting)
+
+        ignored <- NonEmptyMap
+          .fromMap(withoutIgnored.keys.foldLeft(distribution.toSortedMap) {
+            case (acc, nonIgnoredAddress) => acc - nonIgnoredAddress
+          })
+          .toRight[RewardsError](IgnoredAllAddressesDuringWeighting)
+
+        total = withoutIgnored.map(_.value.toLong).reduceLeft(_ + _)
+        perSnapshot = Math.floorDiv(Math.floorDiv(config.monthly.value, 30 * 24 * 60), timeTriggerInterval.toMinutes)
```

Comment:
Is this a correct calculation at all? According to this code the longer the time trigger the lower the rewards (since time trigger is in the denominator). It should be the other way around I think. Probably it works currently only by accident because the time trigger is `1.minutes`, and dividing by `1` or multiplying by `1` doesn't change the outcome.